core/src/main/scala/better/files/File.scala

```diff
@@ -370,8 +371,15 @@ class File private(val path: Path) {
     val algorithm = MessageDigest.getInstance(algorithmName)
     listRelativePaths.toSeq.sorted foreach { relativePath =>
       val file: File = path resolve relativePath
-      val bytes = if (file.isDirectory) relativePath.toString.getBytes else file.loadBytes
-      algorithm.update(bytes)
+      if (file.isDirectory) {
+        val bytes = relativePath.toString.getBytes
+        algorithm.update(bytes)
+      } else {
+        file.inputStream foreach { stream =>
+          val digestStream = new DigestInputStream(stream, algorithm)
+          while (digestStream.read() != -1) {}
```

Comment:
this while loop is O(#of bytes in a file) which would be slow? we should probably use the chunked version here